### DISCUSSION TITLE
Pretty console logs outgoing requests

### DISCUSSION THREAD

**Discussion by KATT** [trpc/trpc#274] [Ideas]
Would be nice if you could optionally log outgoing requests in the client, colour coded by type, with input, & highlight responses.

Especially nice for queries as the input params are urlencoded json, they're quite hard to read by looking at the URL.

**Comment by freeatnet** [comment]
@KATT bit of feedback on this: while it looks very fancy and nice, I noticed two problems with the current (`@trpc/*@10.7`) pretty console logs:

1. In development, I need logging to quickly gleam two things: was the request I'm looking at successful? if so, was there data returned? I feel like the current log structure/pretty formatting aren't particularly conducive to quick reading. As a result, I've implemented a simple logger for myself, which has the following log format: `[<<|>>] [path] #[id] [result or error object] [rest of operation context]`.
2. The simpler logger also works much better for capturing console.logs in Sentry, which does not support message formatting and also tends to cut off lengthy messages.

Happy to pull request the simplified logger — let me know if you're interested.

**Reply by KATT** [reply]
Please create an issue instead - this is a very old discussion so it's just chance that I saw this :)
